File: app/Models/Dto/v1/ServiceAlertsV1Converter.php
<?php

namespace Irail\Models\Dto\v1;

use Irail\Http\Requests\IrailHttpRequest;
use Irail\Http\Requests\ServiceAlertsV1Request;
use Irail\Models\Message;
use Irail\Models\MessageLink;
use Irail\Models\MessageType;
use Irail\Models\Result\LiveboardSearchResult;
use Irail\Models\Result\ServiceAlertsResult;
use stdClass;

class ServiceAlertsV1Converter extends V1Converter
{

    /**
     * @param IrailHttpRequest      $request
     * @param LiveboardSearchResult $result
     */
    public static function convert(ServiceAlertsV1Request $request,
        ServiceAlertsResult $serviceAlerts): DataRoot
    {
        $result = new DataRoot('disturbances');
        $result->disturbance = array_map(fn($alert) => self::convertDisturbance($alert), $serviceAlerts->getAlerts());

        return $result;
    }

    private static function convertDisturbance(Message $alert): stdClass
    {
        $disturbance = new StdClass();
        $disturbance->title = $alert->getHeader();
        $disturbance->description = strip_tags($alert->getMessage());
        $disturbance->type = $alert->getType() == MessageType::WORKS ? 'planned' : 'disturbance';
        $disturbance->link = $alert->getLinks()[0]->getLink();
        $disturbance->timestamp = $alert->getLastModified()->getTimestamp();
        $disturbance->richtext = $alert->getMessage();
        $disturbance->descriptionLink = array_map(fn($link) => self::convertDisturbanceLink($link), $alert->getLinks());
        return $disturbance;
    }

    private static function convertDisturbanceLink(MessageLink $link): stdClass
    {
        $result = new StdClass();
        $result->link = $link->getLink();
        $result->text = $link->getText();
        return $result;
    }

}

File: src/main/java/be/irail/api/legacy/ServiceAlertsV1Converter.java
package be.irail.api.legacy;

import be.irail.api.dto.Message;
import be.irail.api.dto.MessageLink;
import be.irail.api.dto.MessageType;
import be.irail.api.dto.result.ServiceAlertsResult;

/**
 * Converts ServiceAlertsResult to V1 DataRoot format for legacy API compatibility.
 */
public class ServiceAlertsV1Converter extends V1Converter {

    /**
     * Converts a service alerts result to the V1 DataRoot format.
     *
     * @param serviceAlerts the service alerts result
     * @return the DataRoot for V1 output
     */
    public static DataRoot convert(ServiceAlertsResult serviceAlerts) {
        DataRoot result = new DataRoot("disturbances");
        result.disturbance = serviceAlerts.getAlerts().stream()
                .map(ServiceAlertsV1Converter::convertDisturbance)
                .toArray();
        return result;
    }

    private static V1Disturbance convertDisturbance(Message alert) {
        V1Disturbance disturbance = new V1Disturbance();
        disturbance.title = alert.getHeader();
        disturbance.description = stripTags(alert.getMessage());
        disturbance.type = alert.getType() == MessageType.WORKS ? "planned" : "disturbance";
        disturbance.link = alert.getLinks() != null && !alert.getLinks().isEmpty() 
                ? alert.getLinks().getFirst().getLink() : "";
        disturbance.timestamp = alert.getLastModified() != null 
                ? alert.getLastModified().toEpochSecond() : 0;
        disturbance.richtext = alert.getMessage();
        disturbance.descriptionLink = alert.getLinks() != null 
                ? alert.getLinks().stream()
                        .map(ServiceAlertsV1Converter::convertDisturbanceLink)
                        .toArray(V1DescriptionLink[]::new)
                : new V1DescriptionLink[0];
        return disturbance;
    }

    private static V1DescriptionLink convertDisturbanceLink(MessageLink link) {
        V1DescriptionLink result = new V1DescriptionLink();
        result.link = link.getLink();
        result.text = link.getText();
        return result;
    }

    private static String stripTags(String html) {
        if (html == null) return "";
        return html.replaceAll("<[^>]*>", "");
    }

    public static class V1Disturbance {
        public String title;
        public String description;
        public String type;
        public String link;
        public long timestamp;
        public String richtext;
        public V1DescriptionLink[] descriptionLink;
    }

    public static class V1DescriptionLink {
        public String link;
        public String text;
    }
}
